src/main/java/com/synopsys/integration/jenkins/JenkinsVersionHelper.java

```diff
@@ -27,24 +27,23 @@
 import hudson.Plugin;
 import hudson.PluginWrapper;
 import hudson.util.VersionNumber;
-import jenkins.model.Jenkins;
 
 public class JenkinsVersionHelper {
-    private final Jenkins jenkins;
+    private final JenkinsWrapper jenkinsWrapper;
 
-    public JenkinsVersionHelper(Jenkins jenkins) {
-        this.jenkins = jenkins;
+    public JenkinsVersionHelper(JenkinsWrapper jenkinsWrapper) {
+        this.jenkinsWrapper = jenkinsWrapper;
     }
 
     public Optional<String> getPluginVersion(String pluginName) {
-        return Optional.ofNullable(jenkins)
+        return Optional.ofNullable(jenkinsWrapper.getJenkins()).orElse(null)
```

Comment:
This doesn't make sense since the `jenkinsWrapper.getJenkins()` returns an optional. So you don't need to wrap it in another Optional and you don't need the extra `.orElse(null)` the one at the end is fine.